mlflow/pytorch/__init__.py

```diff
@@ -285,6 +285,106 @@ def gen_data():
+        x_data = torch.Tensor([[1.0], [2.0], [3.0]])
+        # Y data with its expected value: labels
+        y_data = torch.Tensor([[2.0], [4.0], [6.0]])
+        # Partial Model example modified from Sung Kim
+        # https://github.com/hunkim/PyTorchZeroToAll
+        class Model(torch.nn.Module):
+            def __init__(self):
+               super().__init__()
+               self.linear = torch.nn.Linear(1, 1)  # One in and one out
+            def forward(self, x):
+                y_pred = self.linear(x)
+            return y_pred
+        # our model
+        model = Model()
+        criterion = torch.nn.MSELoss(size_average=False)
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+        # Training loop
+        for epoch in range(500):
+            # Forward pass: Compute predicted y by passing x to the model
+            y_pred = model(x_data)
+            # Compute and print loss
+            loss = criterion(y_pred, y_data)
+            print(epoch, loss.data.item())
+            #Zero gradients, perform a backward pass, and update the weights.
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+
+        # After training
+        for hv in [4.0, 5.0, 6.0]:
+            hour_var = torch.Tensor([[hv]])
+            y_pred = model(hour_var)
+            print("predict (after training)",  hv, model(hour_var).data[0][0])
+        # log the model
+        with mlflow.start_run() as run:
+            mlflow.log_param("epochs", 500)
+            mlflow.pytorch.log_state_dict(model, "models")
+
+            # logging scripted module
+            scripted_pytorch_model = torch.jit.script(model)
+            mlflow.pytorch.log_state_dict(scripted_pytorch_model, "models")
+    """
+    pickle_module = pickle_module or mlflow_pytorch_pickle_module
+
+    with TempDir() as tmp:
+        local_path = tmp.path("model")
+        save_state_dict(
+            state_dict=state_dict, path=local_path, pickle_module=pickle_module, **kwargs
+        )
+        mlflow.tracking.fluent.log_artifacts(local_path, artifact_path)
+
+
+def save_state_dict(state_dict, path, pickle_module=None, **kwargs):
+    """
+    Save the model as state dict to a path on the local file system
+
+    :param state_dict: state dict of the pytorch model. Ex: model.state_dict()
+    :param path: Local path where the model is to be saved.
+    :param kwargs: kwargs to pass to ``torch.save`` method.
+    """
+    pickle_module = pickle_module or mlflow_pytorch_pickle_module
+
+    import torch
+
+    os.makedirs(path)
+
+    model_data_subpath = "data"
+    model_data_path = os.path.join(path, model_data_subpath)
+    os.makedirs(model_data_path)
+
+    pickle_module_path = os.path.join(model_data_path, _PICKLE_MODULE_INFO_FILE_NAME)
+    with open(pickle_module_path, "w") as f:
+        f.write(pickle_module.__name__)
+
+    model_path = os.path.join(model_data_path, _SERIALIZED_TORCH_MODEL_FILE_NAME)
```

Comment:
A uesr might log a state_dict that represents a checkpoint for inference and/or resuming training ([this use case](https://pytorch.org/tutorials/beginner/saving_loading_models.html#saving-loading-a-general-checkpoint-for-inference-and-or-resuming-training)). In this case `_SERIALIZED_TORCH_MODEL_FILE_NAME` (= `"model.pth"`) doesn't seem to be the right name because it's not a model.

Reply:
Maybe `state_dict.pth` is better?

Reply:
renamed it to `state_dict.pth`